entity-framework/core/what-is-new/ef-core-5.0/breaking-changes.md

```diff
@@ -47,3 +52,129 @@ modelBuilder.Entity<GeoEntity>(
         x.Property(e => e.Point).HasColumnType("POINTZ");
     });
 ```
+
+<a name="required-dependent"></a>
+### Required on the navigation from principal to dependent has different semantics
+
+[Tracking Issue #17286](https://github.com/aspnet/EntityFrameworkCore/issues/17286)
+
+**Old behavior**
+
+Only the navigations to principal could be configured as required. Therefore using `RequiredAttribute` on the navigation to the dependent (the entity containing the foreign key) would instead create the foreign key on the defining entity type.
+
+**New behavior**
+
+With the added support for required dependents it is now possible to mark any reference navigation as required, meaning that in the case shown above the foreign key will be defined on the other side of the relationship and the properties won't be marked as required.
+
+Calling `IsRequired` before specifying the dependent end is now ambiguous:
+
+```cs
+modelBuilder.Entity<Blog>()
+    .HasOne(b => b.BlogImage)
+    .WithOne(i => i.Blog)
+    .IsRequired()
+    .HasForeignKey<BlogImage>(b => b.BlogForeignKey);
+```
+
+**Why**
+
+The new behavior is necessary to enable support for required dependents ([see #12100](https://github.com/dotnet/efcore/issues/12100)).
+
+**Mitigations**
+
+Remove `RequiredAttribute` from the navigation to the dependent and place it instead on the navigation to the principal or configure the relationship in `OnModelCreating`:
+
+```cs
+modelBuilder.Entity<Blog>()
+    .HasOne(b => b.BlogImage)
+    .WithOne(i => i.Blog)
+    .HasForeignKey<BlogImage>(b => b.BlogForeignKey)
+    .IsRequired();
+```
+
+<a name="non-added-generation"></a>
+### Value generators are called when the entity state is changed from Detached to Unchanged, Updated or Deleted
+
+[Tracking Issue #15289](https://github.com/aspnet/EntityFrameworkCore/issues/15289)
+
+**Old behavior**
+
+Value generators were only called when the entity state changed to Added.
+
+**New behavior**
+
+Value generators are now called when the entity state is changed from Detached to Unchanged, Updated or Deleted and the property contains the default values.
+
+**Why**
+
+This change was necessary to improve the experience with properties that are not persisted to the data store and have their value generated always on the client.
+
+**Mitigations**
+
+To prevent the value generator from being called assign a non-default value to the property before the state is changed.
+
+<a name="relational-model"></a>
+### IMigrationsModelDiffer now uses IRelationalModel
+
+[Tracking Issue #20305](https://github.com/aspnet/EntityFrameworkCore/issues/20305)
+
+**Old behavior**
+
+`IMigrationsModelDiffer` API was defined using `IModel`.
+
+**New behavior**
+
+`IMigrationsModelDiffer` API now uses `IRelationalModel`. However the model snapshot still contains only `IModel` as this code is part of the application and Entity Framework can't change it without making a bigger breaking change.
```

Comment:
We still have an issue to update the snapshot, right? I'd really like to do that if we can...

Reply:
You mean https://github.com/dotnet/efcore/issues/18557? It would still be opt-in, so we'll have to support old snapshots for a while.

Reply:
Kind of... We could switch to new snapshots only having a RelationalModel but the default implementation could get it from any existing IModel. That command would just help "normalize" old migrations and snapshots.